### PR TITLE
Add `unstack()` support for non-multiindexed dataframes

### LINKED ISSUE #6694: [FEA]use cuDF dataframe cannot use unstack() function
**Describe the bug**
cuDF dataframe not support .unstack() function

**Steps/Code to reproduce bug**
```python
issue_test_df = cudf.DataFrame({'0':[1,1,1,2,2,3,3,3,3],
                                '1':[1,2,3,1,2,1,2,3,4],
                                '2':[0.2,0.2,0.3,0.3,0.4,0.5,0.6,0.7,0.8]})
issue_test_df.unstack()

---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-63-3fff8526a571> in <module>
----> 1 issue_test_df.unstack()

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/dataframe.py in unstack(self, level, fill_value)
   7061     def unstack(self, level=-1, fill_value=None):
   7062         return cudf.core.reshape.unstack(
-> 7063             self, level=level, fill_value=fill_value
   7064         )
   7065 

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/reshape.py in unstack(df, level, fill_value)
    897     if not isinstance(df.index, cudf.MultiIndex):
    898         raise NotImplementedError(
--> 899             "Calling unstack() on a DataFrame without a MultiIndex "
    900             "is not supported"
    901         )

NotImplementedError: Calling unstack() on a DataFrame without a MultiIndex is not supported
```

**Expected behavior**
Like pandas.DataFrame().unstack(), it should output the unstack table

**Environment overview (please complete the following information)**
 - Environment location: Docker
 - Method of cuDF install: Docker
   - If method of install is [Docker], provide `docker pull` & `docker run` commands used
`sudo docker run --gpus all -it -v /data/project/demo/:/rapids/notebooks/demo/ -p 8889:8888 nvcr.io/nvidia/rapidsai/rapidsai:cuda11.0-runtime-ubuntu18.04`

Please let me know if this is enough to reproduce the issue, thanks !

### PR DESCRIPTION
**Pull request by isVoid**
Closes #6694 

When `unstack()` receives a dataframe with "single" index, returns a series to match pandas behavior.

### CODE REVIEW

File: python/cudf/cudf/core/reshape.py [hunk 1]
```diff
@@ -964,6 +969,21 @@ def unstack(df, level, fill_value=None):
     a
     1     5  <NA>     6  <NA>     7
     2  <NA>     8  <NA>     9  <NA>
+
+    Unstacking single level index dataframe:
+
+    >>> df.unstack(['b', 'd']).unstack()
```
Comment (brandon-b-miller) [inline python/cudf/cudf/core/reshape.py:975]: I think this example is a little opaque - it's sometimes difficult to visualize exactly what the result of `unstack` should be for even a single level, and here I find it a little hard to connect to dots through the chained operation. I'd recommend an example that starts with a dataframe with a single index and shows the result of unstacking that dataframe into a series instead.

File: python/cudf/cudf/core/reshape.py [hunk 2]
```diff
@@ -972,10 +992,16 @@ def unstack(df, level, fill_value=None):
             return df
     df = df.copy(deep=False)
     if not isinstance(df.index, cudf.MultiIndex):
-        raise NotImplementedError(
-            "Calling unstack() on a DataFrame without a MultiIndex "
-            "is not supported"
-        )
+        if isinstance(df, cudf.DataFrame):
+            res = df.T.stack(dropna=False)
```
Comment (brandon-b-miller) [inline python/cudf/cudf/core/reshape.py:996]: Does this pass the typecasting behavior off to transpose? Should we check the dtypes and possibly error here?

**Review by brandon-b-miller — CHANGES_REQUESTED** [review]
Couple of questions otherwise LGTM.

File: python/cudf/cudf/core/reshape.py (hunk 2 above)
Comment (isVoid) [reply python/cudf/cudf/core/reshape.py:996]: It seems like both `transpose.pyx` and `libcudf::transpose.cu` checks whether all columns have the same datatype. A clear exception gets raised if the columns are of different types.

Comment (brandon-b-miller) [reply python/cudf/cudf/core/reshape.py:993]: I would support checking here - imagining what happens here from the user perspective, if I get an error trying to `unstack` a cuDF dataframe, I might wonder why the transpose code is unhappy. 

In general, I think we try and avoid letting libcudf itself serve an error to the user and favor a more surface level python error, usually when I've managed to actually manifest a libcudf error from the python API it means something is very wrong.

File: python/cudf/cudf/core/reshape.py [hunk 3]
```diff
@@ -964,18 +969,48 @@ def unstack(df, level, fill_value=None):
...
             return df
     df = df.copy(deep=False)
     if not isinstance(df.index, cudf.MultiIndex):
-        raise NotImplementedError(
-            "Calling unstack() on a DataFrame without a MultiIndex "
-            "is not supported"
-        )
+        if isinstance(df, cudf.DataFrame):
+            dtype = df._columns[0].dtype
+            if any(
```
Comment (brandon-b-miller) [inline python/cudf/cudf/core/reshape.py:1000]: It won't matter much in most cases but I think it's slightly more efficient to use a single loop that raises `if df._columns[i].dtype != dtype` as opposed to creating a list with all the booleans and then doing an `any` reduction.

**Review by brandon-b-miller — APPROVED** [review]

File: python/cudf/cudf/core/reshape.py (hunk 3 above)
Comment (brandon-b-miller) [reply python/cudf/cudf/core/reshape.py:1000]: Raised an issue since this comes up in a bunch of places https://github.com/rapidsai/cudf/issues/7067

**Comment by isVoid** [comment]
rerun tests

**Pull request merged**